### PR TITLE
Normalize server.js code

### PR DESCRIPTION
**Pull request by lucasefe**
## Description

As a follow up of the conversation regarding code style, and preparing the server.js to accommodate the quota limiting feature, I change the server.js file. 

/cc @zendesk/zendesk-radar 

## Risks

Very low.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -142,45 +142,49 @@ Server.prototype._handleSocketMessage = function(socket, data) {
...
+  
+  this._processMessage(socket, message);
+}
+
+Server.prototype._processMessage = function(socket, message) {
+  var messageType = this._getMessageType(message),
+      resource = this._getResource(message, messageType);
 
-  if (!this._messageAuthorize(message, socket)) {
+  if (!this._authorizeMessage(socket, message, messageType)) {
```
Comment (vanchi-zendesk) [inline server/server.js]: I am sure this is something Pat would like to be left alone. But I realized I preferred verb first after reading through the code. @bolddane What do you think? Is this ok?

File: server/server.js [hunk 2]
```diff
@@ -203,15 +213,20 @@ Server.prototype._messageAuthorize =  function (message, socket) {
...
-    var definition = Type.getByExpression(name);
+Server.prototype._getResource = function(message, messageType) {
+  var name = message.to,
+      type = messageType.type;
 
-    if (definition && Core.Resources[definition.type]) {
-      this.resources[name] = new Core.Resources[definition.type](name, this, definition);
+  if (!this.resources[name]) {
+    if (type && Core.Resources[type]) {
+      this.resources[name] = new Core.Resources[type](name, this, messageType);
```
Comment (vanchi-zendesk) [inline server/server.js]: This is a bug. We should not save it to resources unless we are authorized etc

File: server/server.js [hunk 3]
```diff
@@ -142,45 +142,49 @@ Server.prototype._handleSocketMessage = function(socket, data) {
...
   return true;
 };
 
 // Get a resource, subscribe where required, and handle associated message
-Server.prototype._resourceMessageHandle = function (socket, message) {
-  var resource = this._resourceGet(message.to);
+Server.prototype._handleResourceMessage = function(socket, message, resource) {
   if (resource) {
     logging.info('#socket.message - received', socket.id, message,
       (this.resources[message.to] ? 'exists' : 'not instantiated'),
```
Comment (vanchi-zendesk) [inline server/server.js]: this is an existing bug it seems. It's always going to say 'exists', right?

**Comment by vanchi-zendesk** [comment]
Taking the type-get out is nice. We were doing it twice unnecessarily.

File: server/server.js (hunk 3 above)
Comment (lucasefe) [inline server/server.js]: I didn't change this. Should I? It should probably a different PR.

File: server/server.js (hunk 2 above)
Comment (lucasefe) [inline server/server.js]: I'll fix it.

File: server/server.js (hunk 1 above)
Comment (bolddane) [inline server/server.js]: @vanchi-zendesk - going with "verb first" is fine by me.  It's more of a standard, so I agree with it for the team.

**Comment by lucasefe** [comment]
@vanchi-zendesk the bug should be gone. The code now only stores the resource on Server#resources after checking if the message is OK.

File: server/server.js (hunk 1 above)
Comment (vanchi-zendesk) [inline server/server.js]: ty.

**Comment by vanchi-zendesk** [comment]
:+1:

**Pull request merged**